### PR TITLE
Add mock functions under MockUefiRuntimeServicesTableLib, MockUefiRuntimeServicesTableLib and Create Mock for DxeServicesTableLib

### PR DESCRIPTION
**Pull request by YiTa-AMI**
## Description

Add mock functions under MockUefiRuntimeServicesTableLib, MockUefiRuntimeServicesTableLib and Create Mock for DxeServicesTableLib

For details on how to complete to complete these options and their meaning refer to [CONTRIBUTING.md](https://github.com/microsoft/mu/blob/HEAD/CONTRIBUTING.md).

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Unit tests component can call these mock functions success

## Integration Instructions

N/A

### CODE REVIEW

**Comment by YiTa-AMI** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by YiTa-AMI** [comment]
Hi @VivianNK , @apop5 and @os-d
Would you kindly help review this change?
Thanks,
YiTa

File: MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiDxeServicesTableLib.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/** @file
+  Google Test mocks for DxeServicesTableLib
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_UEFI_DXE_SERVICES_TABLE_LIB_H_
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiDxeServicesTableLib.h]: Naming convention is to use the library name you are mocking, ie
`DxeServicesTableLib` -> `MOCK_DXE_SERVICES_TABLE_LIB_H_` with no `UEFI`

Comment (YiTa-AMI) [reply MdePkg/Test/Mock/Include/GoogleTest/Library/MockUefiDxeServicesTableLib.h]: Change library name from MockUefiDxeServicesTableLib to MockDxeServicesTableLib

**Review by VivianNK — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

**Pull request merged**